packages/react-composites/tests/browser/chat/ChatComposite.test.ts

```diff
@@ -34,18 +59,58 @@ describe('Chat Composite Puppeteer Tests', () => {
   });
 
   test('can load correctly', async () => {
-    await page.waitForSelector(dataUiId(CHAT_UI_IDS.chatScreen), { visible: true });
-    const chatCompositeSelector = await page.$(dataUiId(CHAT_UI_IDS.chatScreen));
+    await pages[0].bringToFront();
+    await pages[0].waitForSelector(dataUiId(CHAT_UI_IDS.chatScreen));
+    const chatCompositeSelector = await pages[0].$(dataUiId(CHAT_UI_IDS.chatScreen));
     expect(chatCompositeSelector).toBeTruthy();
   });
 
   test('can send message', async () => {
-    await page.waitForSelector(dataUiId(COMPONENT_UI_IDS.sendboxTextfield), { visible: true });
-    await page.type(dataUiId(COMPONENT_UI_IDS.sendboxTextfield), 'HELLO WORLD');
-    await page.keyboard.press('Enter');
-    const inner_html = await page.$$eval('.ui-chat__message__content', (nodes) => {
+    await pages[0].bringToFront();
+    await pages[0].waitForSelector(dataUiId(COMPONENT_UI_IDS.sendboxTextfield));
+    await pages[0].type(dataUiId(COMPONENT_UI_IDS.sendboxTextfield), 'How the turn tables');
+    await pages[0].keyboard.press('Enter');
+    await pages[0].waitForSelector(`[data-ui-status="delivered"]`);
+    const inner_html = await pages[0].$$eval(`[data-ui-status="delivered"]`, (nodes) => {
```

Comment:
I'm a bit wary of adding attributes beyond IDs so that the tests can target these elements. Could you (perhaps IRL) quickly explain why we needed this attribute?

Reply:
The good thing about `data-` attributes in html is that they don't affect how the UI renders and are only used for custom purposes.

I have added a logic in message content which changes its data attribute value to its delivery status allowing us to hook into all the delivered messages in a browser.